### ISSUE TITLE
no context menu in sway

### ISSUE THREAD

**Issue by doctorcolossus** [qutebrowser/qutebrowser#6915]
**Version info**:
qutebrowser v2.4.0
Git commit: 
Backend: QtWebEngine 5.15.7, based on Chromium 87.0.4280.144
Qt: 5.15.2

CPython: 3.10.1
PyQt: 5.15.6

sip: 6.1.0.dev2104271705
colorama: 0.4.4
jinja2: 3.0.3
pygments: no
yaml: 5.4.1
adblock: 0.5.1
PyQt5.QtWebEngineWidgets: yes
PyQt5.QtWebEngine: 5.15.5
PyQt5.QtWebKitWidgets: yes
pdf.js: no
sqlite: 3.37.0
QtNetwork SSL: OpenSSL 1.1.1m  14 Dec 2021

Style: QFusionStyle
Platform plugin: wayland
OpenGL: Intel, 4.6 (Compatibility Profile) Mesa 21.2.5
Platform: Linux-5.15.11-arch2-1-x86_64-with-glibc2.33, 64bit
Linux distribution: Arch Linux (arch)
Frozen: False
Imported from /usr/lib/python3.10/site-packages/qutebrowser
Using Python from /usr/bin/python3
Qt library executable path: /usr/lib/qt/libexec, data path: /usr/share/qt

Paths:
cache: /home/casey/.cache/qutebrowser
config: /home/casey/.config/qutebrowser
data: /home/casey/.local/share/qutebrowser
runtime: /run/user/1000/qutebrowser
system data: /usr/share/qutebrowser

**Does the bug happen if you start with `--temp-basedir`?**:

Yes.

**Description**

I do not get any context menu when right-clicking.

I have tried both with my laptop's touchpad, using a two-finger tap, which works in other applications and even in qutebrowser on websites which handle a right-click event (such as [Cronometer](https://cronometer.com/), to delete an item from one's diary), as well as using an actual USB mouse. For both, I see the following events reported by "[mouse event test page](https://unixpapa.com/js/testmouse.html)":

```
mousedown   which=3 button=2 buttons=2
contextmenu which=3 button=2 buttons=2
mouseup     which=3 button=2 buttons=0
```

When right-clicking on an image, I would expect to be presented with a context menu, allowing me to choose to save it to disk. Instead, nothing happens.

When right-clicking on a link, I would expect to see a context-menu where I could select to open it in a new tab or window. Instead, the browser seems to freeze (the link remains underlined despite moving the cursor elsewhere, scrolling does not work with two-finger touchpad scrolling, a mouse wheel, or arrow keys - but qutebrowser commands such as 'o' do still work) until I click somewhere else.

**How to reproduce**

I am not sure how to reproduce, but presumably just use qutebrowser in sway with qt5-wayland? Am I the only one experiencing this issue? Please let me know if I can provide any further details which could help.

**Comment by doctorcolossus** [comment]
After a reboot, I am no longer having this problem...

I will try to remember to follow up if it happens again.

**Comment by doctorcolossus** [comment]
Never mind, it's happening again now.

**Comment by The-Compiler** [comment]
The context menu is entirely handled by the underlying Qt/QtWebEngine, so I doubt that there is much qutebrowser can do about this. Could it be that this only happens when multiple screens are involved, similar to this perhaps? [[QTBUG-93561] Wrong context menu position when attaching external screen - Qt Bug Tracker](https://bugreports.qt.io/browse/QTBUG-93561)

**Comment by The-Compiler** [comment]
@doctorcolossus any update on this?

**Comment by The-Compiler** [comment]
Closing due to inactivity.

**Comment by doctorcolossus** [comment]
Hey @The-Compiler , apologies for missing your replies!

For me this did not involve any external screen. I was just using my laptop's built-in screen.

I have switched to Firefox for now, not because of this issue but because qutebrowser isn't able to suspend tabs. I have very little discipline with tab management, so I would constantly run low on memory. But I suspect these two things might be somehow related, because as I recall, the context menu would work fine when I first started the browser, but stop working after some time, when I probably had too many things open.

If I can provide any more details, let me know and I can give it another try. I will try to pay better attention to my notifications.
